### PR TITLE
feat(planner): support explain ast

### LINKED ISSUE #7063: Feature: explain AST/syntax statement
**Summary**

Description for this feature.

- explain syntax:  output the formatted SQL .

Example:
```
EXPLAIN SYNTAX
SELECT number
FROM numbers(10)

┌─explain──────────┐
│ SELECT number    │
│ FROM numbers(10) │
└──────────────────┘
```	

- explain ast: output the AST struct.

Example:
```
explain ast select number from numbers(10);


┌─explain─────────────────────────────────────┐
│ SelectWithUnionQuery (children 1)           │
│  ExpressionList (children 1)                │
│   SelectQuery (children 2)                  │
│    ExpressionList (children 1)              │
│     Identifier number                       │
│    TablesInSelectQuery (children 1)         │
│     TablesInSelectQueryElement (children 1) │
│      TableExpression (children 1)           │
│       Function numbers (children 1)         │
│        ExpressionList (children 1)          │
│         Literal UInt64_10                   │
└─────────────────────────────────────────────┘
```

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Support `explain ast`.
It is not easy to implement `CockroachDB` style explain format with `pretty`, so I still use `FormatTreeNode`.

Fixes #7063

### CODE REVIEW

**Comment by Xuanwo** [comment]
It seems like we have to maintian three kinds of `Display` for `stmt`. Do we have better choices?

**Comment by b41sh** [comment]
> It seems like we have to maintian three kinds of `Display` for `stmt`. Do we have better choices?

It is indeed a problem. If a `statement` is changed, we need to modify three places.

**Comment by drmingdrmer** [comment]
- `main` branch is force pushed, update your working branch please: #7261

**Comment by drmingdrmer** [comment]
`main` branch is force pushed(**again** `:(`, to cleanup big object in history), update your working branch please: [To everybody: main branch is force pushed. #7263](https://github.com/datafuselabs/databend/issues/7263)

**Comment by Xuanwo** [comment]
> It is indeed a problem. If a `statement` is changed, we need to modify three places.

Is it possible to automatically convert the statement with the same rules instead of writing similar code three times?

**Comment by sundy-li** [comment]
I did not find better solution for this.  cc @leiysky  do you have any suggestions?

**Comment by b41sh** [comment]
> > It is indeed a problem. If a `statement` is changed, we need to modify three places.
> 
> Is it possible to automatically convert the statement with the same rules instead of writing similar code three times?

It is not easy to write a transfer rule, because every statement has different fields.
`Clickhouse` does this by adding a `formatImpl` function to each statement.
https://github.com/ClickHouse/ClickHouse/blob/master/src/Parsers/IAST.h#L231
Although there are many statements, they are relatively few changes, we can check and modify them regularly.

File: src/query/ast/src/ast/format/ast/mod.rs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+use crate::ast::ExplainKind;
+use crate::ast::FormatTreeNode;
+use crate::ast::Statement;
+
+pub fn format_statement(stmt: Statement) -> Result<String> {
+    format_stmt(stmt).format_pretty()
+}
+
+fn format_stmt(stmt: Statement) -> FormatTreeNode<String> {
```
Comment (sundy-li) [inline src/query/ast/src/ast/format/ast/mod.rs:37]: can we use a common visitor to apply the `format` ?

Comment (b41sh) [reply src/query/ast/src/ast/format/ast/mod.rs:37]: ok

**Review by sundy-li — APPROVED** [review]

**Comment by b41sh** [comment]
I implemented `trait Visitor`, if there are new `Statement` in the future, we only need to add the corresponding visit method. PTAL @Xuanwo

**Review by Xuanwo — APPROVED** [review]
Great!

**Comment by BohuTANG** [comment]
cc @soyeric128 for documentation :)

```
explain ast
```

**Pull request merged**